drivers/include/lis2dh12.h

```diff
@@ -114,13 +114,61 @@ enum {
     LIS2DH12_OK    =  0,            /**< everything was fine */
     LIS2DH12_NOBUS = -1,            /**< bus interface error */
     LIS2DH12_NODEV = -2,            /**< unable to talk to device */
+    LIS2DH12_NOINT = -3,            /**< wrong interrupt line was given (1 or 2) */
 };
 
+/**
+ * @brief    Parameter for interrupt configuration
+ */
+typedef struct {
+    uint8_t int_config;                 /**< values for configuration */
+    uint8_t int_threshold:7;            /**< the threshold for triggering interrupt, threshold in range 0-127 */
+    uint8_t int_duration:7;             /**< time between two interrupts, calc from <duration>/ODR, ODR section in CTRL_REG1, duration in range 0-127 */
+    uint8_t int_type;                   /**< values for type of interrupts */
+} int_params_t;
+
+/**
+ * @brief   Status of INT_SRC register
+ */
+typedef struct {
+    uint8_t LIS2DH12_INT_SRC_XL:1;  /**<  */
+    uint8_t LIS2DH12_INT_SRC_XH:1;  /**<  */
+    uint8_t LIS2DH12_INT_SRC_YL:1;  /**<  */
+    uint8_t LIS2DH12_INT_SRC_YH:1;  /**<  */
+    uint8_t LIS2DH12_INT_SRC_ZL:1;  /**<  */
+    uint8_t LIS2DH12_INT_SRC_ZH:1;  /**<  */
+    uint8_t LIS2DH12_INT_SRC_IA:1;  /**<  */
+} int_src_reg_t;
```

Comment:
As for `int_params_t` I would suggest prefixing this name with the module name, e.g. `lis2dh12_int_params_t`.